db/view/view.cc

```diff
@@ -1226,6 +1237,18 @@ future<> mutate_MV(
         auto remote_endpoints = service::get_local_storage_proxy().get_token_metadata_ptr()->pending_endpoints_for(view_token, keyspace_name);
         auto sem_units = pending_view_updates.split(mut.fm.representation().size());
 
+        // If a view is marked with the synchronous_updates property, we should wait for all.
+        const bool apply_update_synchronously = [&] {
+            bool update_synchronously = should_update_synchronously(*mut.s);
+            if (update_synchronously) {
+                tracing::trace(tr_state, "Forcing {}.{} view update to be synchronous (synchronous_updates property was set)",
+                    mut.s->ks_name(), mut.s->cf_name()
+                );
+            }
+
+            return wait_for_all || update_synchronously;
+        } ();
```

Comment:
Why is this a lambda called immediately, instead of just staightforward code which ends with something like

const bool apply_update_sychronously = wait_for_all || update_synchronously

A lambda is useful when you call it multiple times. Here you only call it once... I also got confused by this initially because I thought the variable held the lambda, while it holds the result of it.

Reply:
I had the impression that a lot of similar sounding variables (`apply_update_synchronously`, `update_synchronously`) in the same scope could cause confusion, so I hid the `update_synchronously` variable in its own block. I agree that instantly called lambda is not the cleanest solution, I will change it to what you proposed.